### PR TITLE
Add api for available metrics for given plan

### PR DESCRIPTION
**Pull request by tspenov**
image

* Product defaults to API
image

image

### CODE REVIEW

File: lib/sanbase/billing/plan/access_checker.ex [hunk 1]
```diff
@@ -123,6 +123,21 @@ defmodule Sanbase.Billing.Plan.AccessChecker do
...
+  def get_available_metrics_for_plan(product, plan, restriction_type \\ :all) do
+    case restriction_type do
+      :free -> @free_metrics
+      :restricted -> @restricted_metrics
+      :custom -> @custom_access_queries
+      :all -> @all_metrics
+    end
+    |> Enum.filter(&match?({:metric, _}, &1))
+    |> Enum.filter(&plan_has_access?(plan, product, &1))
+    |> Enum.map(fn {_, name} -> name end)
```
Comment (IvanIvanoff) [inline lib/sanbase/billing/plan/access_checker.ex:137]: You can make these 3 `Stream` instead of `map`

**Comment by josefansinger** [comment]
@tspenov Can you reconstruct `plan` from the API key if `plan = null`?

**Comment by tspenov** [comment]
@josefansinger yes, but this query is for customers that don't have an api key yet and are trying to evaluate what metrics does this plan has ?
Otherwise It should be smth like this:
```graphql
currentUser {
  availableMetrics
}
```

**Comment by josefansinger** [comment]
@tspenov Ok. I am currently only checking [available slugs](https://github.com/josefansinger/sanapiwrapper/blob/master/R/sanapiwrapper.R#L194) in my wrapper. Hope this list of slugs is empty, if the metric is not available for the current user/plan/API key. But that's unrelated to the PR.

**Pull request merged**